src/cli.rs

```diff
@@ -1,109 +1,170 @@
-use std::path::PathBuf;
-use anyhow::anyhow;
+use crate::installer::{
+    self, ClientInstallation, LoaderVersion, MinecraftVersion, ServerInstallation,
+};
+use anyhow::Context;
+use anyhow::Result;
 use clap::{Parser, Subcommand};
+use derive_more::Display;
 use reqwest::Client;
-use crate::installer;
-use crate::installer::{ClientInstallation, LoaderVersion, MinecraftVersion};
+use std::path::PathBuf;
 
 #[derive(Parser)]
-#[clap(about, version)]
-#[clap(propagate_version = true)]
+#[command(about, version, propagate_version = true)]
 pub struct Args {
     #[clap(subcommand)]
-    pub subcommand: Option<SubCommands>,
-}
-
-#[derive(Subcommand)]
-pub enum SubCommands {
-    Install {
-        #[command(subcommand)]
-        subcommand: InstallSubcommands
-    }
+    pub subcommand: Option<Subcommands>,
+    /// The Minecraft version to install
+    ///
+    /// Pick between the
+    /// latest `stable` version,
+    /// latest `snapshot`,
+    /// or a specific version number.
+    #[arg(short = 'm', long, default_value_t)]
+    minecraft_version: MCVersionCLI,
+    /// The Quilt loader version to install
+    ///
+    /// Pick between the
+    /// latest `stable` version,
+    /// latest `beta`,
+    /// or a specific version number.
+    #[arg(short = 'l', long, default_value_t)]
+    loader_version: LoaderVersionCLI,
 }
 
 #[derive(Subcommand)]
-pub enum InstallSubcommands {
+pub enum Subcommands {
+    /// Install the Quilt Loader client
     Client {
-        #[arg(short='M',long)]
-        minecraft_version: Option<String>,
-
-        #[arg(short='L',long)]
-        loader_version: Option<String>,
-        #[arg(short='s',long)]
-        snapshot: bool,
-        #[arg(short='b',long)]
-        loader_beta: bool,
-
-        #[arg(short, long)]
+        /// Don't create a profile
+        #[arg(short = 'p', long)]
```

Comment:
While you're at it:
I think doing this with `--create-profile false` (or `-p false`), with `true` being the default value would be more intuitive.

Reply:
That's what I did initially, however that means that if you want to disable profile creation, you'd have to do `--create-profile false` rather than `--no-profile`. Basically I feel like the false is redundant given that true is the default anyways. Also CLI programs generally follow the latter pattern for the same reason.